Konscious.Security.Cryptography.Argon2/Argon2Lane.cs

```diff
@@ -18,18 +20,45 @@ public Argon2Lane(int blockCount)
                     throw new ArgumentOutOfRangeException(nameof(index));
                 }
 
-                return new Argon2Memory(_memory, 128 * index);
+                return _memory.Slice(128*index, 128);
             }
         }
 
-        public int BlockCount
+        public int BlockCount { get; }
+
+        private readonly Memory<ulong> _memory;
+    }
+
+    internal static class SpanExtensions
+    {
+        public static void Blit(this Span<ulong> toBlit, ReadOnlySpan<byte> bytes, int destOffset = 0, int srcOffset = 0, int byteLength = -1)
         {
-            get
+            int remainder = 0;
+            int length;
+            if (byteLength < 0)
             {
-                return _memory.Length / 128;
+                length = 128;
             }
-        }
+            else
+            {
+                length = byteLength / 8;
+                remainder = byteLength - (length * 8);
+            }
+            //---------------
+            var newSpan = MemoryMarshal.Cast<byte, ulong>(bytes[srcOffset..]);
```

Comment:
Is `length` is supposed to be used here to limit the amount taken from `bytes`?

Reply:
No in the previous blit implmentation it was just the total number of whole 8byte chunks

Reply:
It looks they were also using min(data.Length / 8, length) to limit the length of the data moved though doesn't it?

Reply:
For anyone else, we took the `min(data.Length / 8, length)` out of the original impl and all the tests still passed so it's unclear whether it's necessary. Conversation is ongoing about what it does, if we need to do it, and how for anyone interested.

Reply:
resolved

Here's a summary of the convo/resolution for anyone who missed it:

The upstream function accepted source-offset and destination-offset parameters to control the positions in the source and destination arrays that data was read from and written to, and a length parameter to control how much data was moved. The first iteration of the new implementation respected the offset parameters but didn't use the length parameter to constrain how much data was copied, instead relying on the amount of data available after the source offset. The upstream test suite didn't rely on the length constraint and continued to pass using the first iteration of the new implementation, begging the question of whether the use of the function (by the lib) even _could_ require said constraint. Ultimately the source-offset and length parameters were removed in favor of requiring the called to pass precisely the source data they want written to the destination and the supporting change was made at each call site.